### PR TITLE
[KOGITO-4333] - Hardening Kogito Operator permissions

### PR DESCRIPTION
**Pull request by Kaitou786**
See: https://issues.redhat.com/browse/KOGITO-4333

Signed-off-by: Tarun Khandelwal <tarkhand@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

File: bundle/manifests/kogito-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -255,20 +256,30 @@ spec:
...
           - get
           - list
+          - patch
           - update
           - watch
         - apiGroups:
           - ""
           resources:
-          - '*'
+          - configmaps
```
Comment (sutaakar) [inline bundle/manifests/kogito-operator.clusterserviceversion.yaml:278]: This group is same as above, the only difference is `serviceaccounts` resource, could they be merged to one?

**Review by sutaakar — APPROVED** [review]

**Review by vaibhavjainwiz — APPROVED** [review]
LGTM

**Review by radtriste — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**